### ISSUE TITLE
Password Reset

### ISSUE THREAD

**Issue by iliajie** [webmin/webmin#1981]
Jamie, it would be great for Webmin to naturally handle password reset/recovery for every Webmin user, rather than using additional modules, like **Virtualmin Password Recovery** module.

**Comment by iliajie** [comment]
.

**Comment by jcameron** [comment]
I'm not sure we could ever do this without Virtualmin though, otherwise where else are we going to get the recovery email from?

**Comment by iliajie** [comment]
We are going to provide a new _Email_ filed in **Webmin Users** module for each Webmin user.

If a user without email tries to reset the password the error message would say something like: **User doesn't have recovery email address set.** If it's Virtualmin user we can check in Virtualmin domain config for the email address.

**Comment by jcameron** [comment]
Sure, we could do that. I wonder if this is a really highly demanded feature though? It makes a lot of sense for Virtualmin where domain owners don't have root access, but less so for a regular Webmin install that only has a single `root` user who could SSH in and reset the password anyway..

**Comment by iliajie** [comment]
I think this will be on the high demand. Password reset is very core like feature, that in my opinion should be clearly and nicely implemented on Webmin side.

`root` users won't enable password reset (most likely for security reasons). But any other normal user would want to be able to reset their Webmin password without extra hassle, in similar way we change the expired passwords.

**Comment by jcameron** [comment]
Right, but what I mean is most systems with multiple non-root users are running Virtualmin and so already has this reset functionality.

**Comment by iliajie** [comment]
> most systems with multiple non-root users are running Virtualmin and so already has this reset functionality.

They don't, as we don't install that plugin by default. Besides, it  looks bulky and obsolete, and also theming for it doesn't work. Password reset functionality should be a core feature. Should also work for Cloudmin, and it makes no sense to install that plugin for Cloudmin users.

I think we really should implement it!

**Comment by jcameron** [comment]
I'll look into it, but not right now ..

**Comment by iliajie** [comment]
> I'll look into it, but not right now ..

Cool! Thanks!

**Comment by iliajie** [comment]
Jamie, we really need this functionality! Could you take a look? If you're not willing to, I can take a look then ..

**Comment by jcameron** [comment]
I do want to add this but haven't had time yet...
